test/list.test.js

```diff
@@ -2,34 +2,27 @@
 
 const list = require('../list')
 const FtpServer = require('./support/FtpServer')
+const SftpServer = require('./support/SftpServer')
+const { withServer } = require('./support/server')
 const getStream = require('get-stream')
+const each = require('jest-each').default
 
 describe('list', () => {
   it('is a function', () => {
     expect(typeof list).toBe('function')
   })
 
-  it('lists files from the given directory with anonymous user', async () => {
-    const server = new FtpServer()
-    await server.start()
-
-    const stream = await list({ pathname: 'data', ...server.options })
-    const filenames = await getStream.array(stream)
-
-    await server.stop()
-
-    expect(filenames).toEqual(['data/abc.txt', 'data/xyz.txt'])
-  })
-
-  it('lists files from the given directory with user/password', async () => {
-    const server = new FtpServer({ user: 'test', password: '1234' })
-    await server.start()
-
-    const stream = await list({ pathname: 'data', ...server.options })
-    const filenames = await getStream.array(stream)
-
-    await server.stop()
-
-    expect(filenames).toEqual(['data/abc.txt', 'data/xyz.txt'])
+  each([
```

Comment:
Good idea to remove redundancy, but I don't think the `jest-each` package is required and the labels can be more readable:

```js
[{
  label: 'on a ftp server with anonymous user',
  factory: () => new FtpServer()
}].forEach(({ label, factory } => {
  it(`lists files from the given directory ${label}`, () => {
    // ...
  })
})
```